src/yang/example-compa-diffserv.yang

```diff
@@ -32,60 +32,73 @@ module example-compa-diffserv {
      companyA diffserv specification extension.";
   revision YYYY-MM-DD {
     description
-      "Initial revision for diffserv actions on network packets";
+      "Initial revision for diffserv actions on network packets.";
     reference
       "RFC XXXX: A YANG Data Model for Quality of Service (QoS).";
   }
 
   identity default-policy-type {
     base qos-types:policy-type;
     description
-      "This defines default policy-type";
+      "This defines default policy-type.";
   }
 
   identity qos-group {
     base qos-types:filter-type;
     description
-      "qos-group filter-type";
-  }
-
-  identity drop {
-    base qos-types:action-type;
-    description
-      "drop action-type";
+      "Filter-type to match on  packet metadata.";
   }
 
   grouping qos-group {
     list qos-group {
-      key "qos-group-min qos-group-max";
+      key "min max";
       description
-        "list of dscp ranges";
-      leaf qos-group-min {
+        "List of qos-group ranges.";
+      leaf min {
         type uint8;
         description
-          "Minimum value of qos-group range";
+          "Minimum value of qos-group range.";
       }
-      leaf qos-group-max {
+      leaf max {
         type uint8;
+        must ". >= ../min" {
+          error-message
+          "The qos-group max must be 
+           greater than or equal to qos-group min.";
+        }
         description
-          "Maximum value of qos-group range";
+          "Maximum value of qos-group range.";
       }
     }
     description
-      "Filter containing list of qos-group ranges";
+      "Filter containing list of qos-group ranges.";
   }
-  grouping drop {
-    container drop {
-      leaf drop {
-        type empty;
+  grouping bw-remaining-percent {
+    container bw-remaining-percent {
+      leaf value {
+        type uint8;
         description
-          "drop action";
+          "Value for the rate that is configured.";
+      }
+        description
+          "Remaining bandwidth of the link in percent.";
+    }
+    description
+      "Grouping for bandwidth remaining after
+      reserved and priority bandwidth.";
+  }
+  grouping qos-group-marking {
+    container qos-group {
+      leaf qos-group {
+        type uint8;
+        description
+          "Mark metadata information in the network packet";
       }
       description
-        "drop container";
+        "Container for Qos-Group marking.";
```

Comment:
I am confused about what is QoS group, so once you have explained somewhere what it is, this may need an update.

Also can we be consistent in using QoS rather than Qos.

Reply:
QoS group is cisco specific packet metadata which can have value 0-255.

Reply:
Ok. Then let us explain that in the example, without mentioning Cisco of course. Also which specific data in the packet is being looked at for the value of 0-255?

Reply:
it is just a tag

Reply:
I assume it is a tag that is carried separately from the packet?? Let us say that somewhere in the example.

Reply:
QoS group is simply set of packet tagged as QoS group.